src/Corax/Queries/AllEntriesMatch.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System;
+using Voron;
+using Voron.Data.Containers;
+using Voron.Data.Sets;
+using Voron.Global;
+using Voron.Impl;
+
+namespace Corax.Queries
+{
+    public struct AllEntriesMatch : IQueryMatch
+    {
+        private readonly Transaction _tx;
+        private readonly long _count;
+        private Set.Iterator _entriesPagesIt;
+        private int _offset;
+        private Page _currentPage;
+        private long _entriesContainerId;
+
+        public unsafe AllEntriesMatch(Transaction tx)
+        {
+            _tx = tx;
+            _count = tx.LowLevelTransaction.RootObjects.ReadInt64(IndexWriter.NumberOfEntriesSlice) ?? 0;
+            _entriesContainerId = tx.OpenContainer(IndexWriter.EntriesContainerSlice);
+            _entriesPagesIt = Container.GetAllPagesSet(tx.LowLevelTransaction, _entriesContainerId).Iterate();
+            _offset = 0;
+            _currentPage = new Page(null);
+        }
+
+        public long Count => _count;
+        public QueryCountConfidence Confidence => QueryCountConfidence.High;
+        public unsafe int Fill(Span<long> matches)
+        {
+            while (true)
```

Comment:
Why do we need that while(true) loop? I think it's executed only once anyway (return statement is inside that loop)

Reply:
You may read all the ids in a page, but still can have space in matches, so we'll move to the next page.